File: Resources/samples/sample.vb
' AI generated sample

Public Class Bank
    Private _name As String
    Private _accounts As Dictionary(Of Integer, Account)

    Public Sub New(name As String)
        _name = name
        _accounts = New Dictionary(Of Integer, Account)()
    End Sub

    Public Function CreateAccount(customer As Customer, initialDeposit As Decimal) As Integer
        Dim account As New Account(customer, initialDeposit)
        _accounts.Add(account.AccountNumber, account)
        Return account.AccountNumber
    End Function

    Public Sub Deposit(accountNumber As Integer, amount As Decimal)
        If _accounts.ContainsKey(accountNumber) Then
            _accounts(accountNumber).Deposit(amount)
        Else
            Console.WriteLine("Account not found.")
        End If
    End Sub

    Public Sub Withdraw(accountNumber As Integer, amount As Decimal)
        If _accounts.ContainsKey(accountNumber) Then
            _accounts(accountNumber).Withdraw(amount)
        Else
            Console.WriteLine("Account not found.")
        End If
    End Sub

    Public Sub DisplayAccountDetails(accountNumber As Integer)
        If _accounts.ContainsKey(accountNumber) Then
            _accounts(accountNumber).DisplayDetails()
        Else
            Console.WriteLine("Account not found.")
        End If
    End Sub
End Class

Public Class Account
    Private Shared _accountNumberCounter As Integer = 1000
    Public ReadOnly Property AccountNumber As Integer
    Private _customer As Customer
    Private _balance As Decimal

    Public Sub New(customer As Customer, initialDeposit As Decimal)
        _customer = customer
        _balance = initialDeposit
        AccountNumber = System.Threading.Interlocked.Increment(_accountNumberCounter)
    End Sub

    Public Sub Deposit(amount As Decimal)
        _balance += amount
        Console.WriteLine("Deposited " & amount & ". New balance: " & _balance)
    End Sub

    Public Sub Withdraw(amount As Decimal)
        If amount > _balance Then
            Console.WriteLine("Insufficient funds.")
        Else
            _balance -= amount
            Console.WriteLine("Withdrew " & amount & ". New balance: " & _balance)
        End If
    End Sub

    Public Sub DisplayDetails()
        Console.WriteLine("Account Number: " & AccountNumber)
        Console.WriteLine("Customer Name: " & _customer.Name)
        Console.WriteLine("Account Balance: " & _balance)
    End Sub
End Class

Public Class Customer
    Public ReadOnly Property Name As String
    Private _address As String

    Public Sub New(name As String, address As String)
        Me.Name = name
        _address = address
    End Sub
End Class

' Example usage:
Module Module1
    Sub Main()
        Dim bank As New Bank("Copilot Bank")

        ' Create customers
        Dim john As New Customer("John Doe", "123 Elm Street")
        Dim jane As New Customer("Jane Doe", "456 Oak Street")

        ' Create accounts
        Dim johnsAccount As Integer = bank.CreateAccount(john, 500)
        Dim janesAccount As Integer = bank.CreateAccount(jane, 1000)

        ' Make some transactions
        bank.Deposit(johnsAccount, 300)
        bank.Withdraw(johnsAccount, 200)
        bank.Deposit(janesAccount, 500)
        bank.Withdraw(janesAccount, 1500)

        ' Display account details
        bank.DisplayAccountDetails(johnsAccount)
        bank.DisplayAccountDetails(janesAccount)
    End Sub
End Module

File: Resources/samples/sample.cpp
// AI generated sample

#include <iostream>
#include <string>
#include <map>

class Customer {
private:
    std::string name;
    std::string address;

public:
    Customer(const std::string& name, const std::string& address) : name(name), address(address) {}

    std::string getName() const {
        return name;
    }
};

class Account {
private:
    static int accountNumberCounter;
    int accountNumber;
    Customer customer;
    double balance;

public:
    Account(const Customer& customer, double initialDeposit) : customer(customer), balance(initialDeposit) {
        accountNumber = ++accountNumberCounter;
    }

    void deposit(double amount) {
        balance += amount;
        std::cout << "Deposited " << amount << ". New balance: " << balance << std::endl;
    }

    void withdraw(double amount) {
        if (amount > balance) {
            std::cout << "Insufficient funds." << std::endl;
        } else {
            balance -= amount;
            std::cout << "Withdrew " << amount << ". New balance: " << balance << std::endl;
        }
    }

    void displayDetails() const {
        std::cout << "Account Number: " << accountNumber << std::endl;
        std::cout << "Customer Name: " << customer.getName() << std::endl;
        std::cout << "Account Balance: " << balance << std::endl;
    }

    int getAccountNumber() const {
        return accountNumber;
    }
};

int Account::accountNumberCounter = 1000;

class Bank {
private:
    std::string name;
    std::map<int, Account> accounts;

public:
    Bank(const std::string& name) : name(name) {}

    int createAccount(const Customer& customer, double initialDeposit) {
        Account account(customer, initialDeposit);
        accounts[account.getAccountNumber()] = account;
        return account.getAccountNumber();
    }

    void deposit(int accountNumber, double amount) {
        if (accounts.find(accountNumber) != accounts.end()) {
            accounts[accountNumber].deposit(amount);
        } else {
            std::cout << "Account not found." << std::endl;
        }
    }

    void withdraw(int accountNumber, double amount) {
        if (accounts.find(accountNumber) != accounts.end()) {
            accounts[accountNumber].withdraw(amount);
        } else {
            std::cout << "Account not found." << std::endl;
        }
    }

    void displayAccountDetails(int accountNumber) {
        if (accounts.find(accountNumber) != accounts.end()) {
            accounts[accountNumber].displayDetails();
        } else {
            std::cout << "Account not found." << std::endl;
        }
    }
};

// Example usage:
int main() {
    Bank bank("Copilot Bank");

    // Create customers
    Customer john("John Doe", "123 Elm Street");
    Customer jane("Jane Doe", "456 Oak Street");

    // Create accounts
    int johnsAccount = bank.createAccount(john, 500);
    int janesAccount = bank.createAccount(jane, 1000);

    // Make some transactions
    bank.deposit(johnsAccount, 300);
    bank.withdraw(johnsAccount, 200);
    bank.deposit(janesAccount, 500);
    bank.withdraw(janesAccount, 1500);

    // Display account details
    bank.displayAccountDetails(johnsAccount);
    bank.displayAccountDetails(janesAccount);

    return 0;
}
